sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1687,6 +1687,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val LEGACY_PASS_PARTITION_BY_AS_OPTIONS =
+    buildConf("spark.sql.legacy.sources.write.passPartitionByAsOptions")
+      .internal()
```

Comment:
Are you sure this must be internal?

I see some of the previous ones with `LEGACY_` prefix are internal but before them there are a few externals: `LEGACY_REPLACE_DATABRICKS_SPARK_AVRO_ENABLED` and `LEGACY_SIZE_OF_NULL`. So what is the reason defining this as internal? 

I assume if we expect the users to set it it then it must be external. 
What is your (and the others) opinion?

Reply:
We don't want to document this flag for users. This is just in case we break some existing production workloads. For other users, they don't need to know anything about this flag.

Reply:
Ok, I see. Thanks for the explanation.